src/corelibs/U2Gui/src/util/project/ProjectTreeController.cpp

```diff
@@ -792,8 +795,64 @@ void ProjectTreeController::sl_onRemoveSelectedItems() {
     const QList<Document *> selectedDocs = getDocsInSelection(deriveDocsFromObjs).values();
     const QList<Folder> selectedFolders = getSelectedFolders();
     const QList<GObject *> selectedObjects = objectSelection.getSelectedObjects();
+    
+    QStringList selectedItemNames;
+    foreach(Document * doc, selectedDocs) {
+        selectedItemNames.append(doc->getName());
+    }
+    foreach(const Folder& folder, selectedFolders) {
+        selectedItemNames.append(folder.getFolderName());
+    }
+    foreach(GObject* obj, selectedObjects) {
+        selectedItemNames.append(obj->getGObjectName());
+    }
+    
+    QString fullItemsNamesList;
+
+    QString warningMessageText = (1 == selectedItemNames.count()) ? tr("Remove selected item?") : tr("Remove selected items?");
```

Comment:
Yoda style comparison is not necessary. gcc compiler will catch and show warnings for lines with assignments in expression.
More: you can't assign here at all (to list.count())

Reply:
Nobody reads compiler warnings.

Reply:
Here what we have in our build file for those who don't like to read warnings like these:
QMAKE_CXXFLAGS += -Werror=parentheses :)